static/examples/audio.html

```diff
@@ -0,0 +1,188 @@
+        * | Event Target | <--- | AudioContext |
+        * |  ____________|      |______________|
+         */
+
+        let distortion = audioCtx.createWaveShaper();
+        let gainNode = audioCtx.createGain();
+        let biquadFilter = audioCtx.createBiquadFilter();
+        let delay = audioCtx.createDelay();
+
+        // distortion curve for the waveshaper, thanks to Kevin Ennis
+        // http://stackoverflow.com/questions/22312841/waveshaper-node-in-webaudio-how-to-emulate-distortion
+        // 
+        // it would be cool to change this into a vocoder / DSP for autotuning
+        // http://blogs.zynaptiq.com/bernsee/pitch-shifting-using-the-ft/
+        // see also https://github.com/k15z/pitch-perfect-auto-tune
+
+        function makeDistortionCurve(amount) {
+          let k = typeof amount === 'number' ? amount : 50,
+            n_samples = 44100,
+            curve = new Float32Array(n_samples),
+            deg = Math.PI / 180,
+            i = 0,
+            x;
+          for ( ; i < n_samples; ++i ) {
+            x = i * 2 / n_samples - 1;
+            curve[i] = ( 3 + k ) * x * 20 * deg / ( Math.PI + k * Math.abs(x) );
+          }
+          return curve;
+        };
+
+
+        if (navigator.mediaDevices.getUserMedia) {
+           const constraints = {
+            audio: {
+              echoCancellation: true,
+              noiseSupression: false,
+              volume: {
+                exact: {
+                  min: 0,
+                  max: 90,
+                },
+                ideal: {
+                  min: 0,
+                  max: 90,
+                }
+              }
+            }
+          }
+          navigator.mediaDevices.getUserMedia(constraints)
+            .then(function(stream) {
+              source = audioCtx.createMediaStreamSource(stream);
+              source.connect(distortion);
+              distortion.connect(biquadFilter);
+              biquadFilter.connect(gainNode);
+              gainNode.connect(delay);
+              delay.connect(audioCtx.destination);
+
+              range.oninput = function() {
+                gainNode.gain.value = range.value;
+              }
+              // Delay time needs to between 0, 1 but the slider only returns integers
+              delayrange.oninput = function() {
+                delay.delayTime.value = delayrange.value / 10;
+              }
+              voiceChange();
+            })
+            .catch( function(err) { console.log('The following gUM error occured: ' + err);})
+        } else {
+          console.log('getUserMedia not supported on your browser!');
+        }
+
+        function voiceChange() {
+
+          distortion.oversample = '4x';
+          biquadFilter.gain.setTargetAtTime(0, audioCtx.currentTime, 0)
+
+          let voiceSetting = voiceSelect.value;
+
+            biquadFilter.disconnect(0);
+            biquadFilter.connect(gainNode);
+
+          if(voiceSetting == "distortion") {
```

Comment:
Minor JS issues in this area, perhaps cuz this code was copied from elsewhere. These issues will be flagged by `eslint` once we get that up and working. 

- Indentation is inconsistent. The `biquad` lines above should be aligned.
- `if(...)` should be `if (...)`, because control constructs are distinct from functions (at least, in JS) and they should not look like function invocations (in my opinion).
- Single quotes for strings in JS.
- Aren't we supposed to be using `===` instead of `==` as a way to be explicit about what type coercion we expect? Or is that not necessary anymore? When I see `==` I assume that it is *classic* JS from before HTML5, and am skeptical of it.